### PR TITLE
fail2ban cleanup

### PR DESCRIPTION
**Pull request by casperklein**
# Description

This is the first part from #1821. This is a cleanup / reorganization, so no breaking change is expected.

The file `target/fail2ban/jail.conf` has been deleted. Now the more recent debian `jail.conf` is used.
The default customization for this project takes now place in `target/fail2ban/jail.local`.
User customization from `/tmp/docker-mailserver/fail2ban-jail.cf` is now reflected in `/etc/fail2ban/jail.d/user-jail.local`.

The file `target/fail2ban/filter.d/postfix-sasl.conf` has been deleted, because its no longer needed. I guess, the file was used in the past, because fail2ban had no default config for postfix-sasl? In my manual tests, bans worked perfect for postfix-sasl. Also all bat tests passed successfully.

The file `config/fail2ban-fail2ban.cf` has been updated with the current debian configuration as example.
The file `config/fail2ban-jail.cf` has been updated with the current debian defaults.

On debian, the sshd jail is activated automatically. As we don't use sshd at all, I've disabled this jail and adjusted the corresponding bat test.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (non-breaking change that does improve existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (README.md or ENVIRONMENT.md or the documentation)
- [ ] If necessary I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by wernerfred** [comment]
Are Updates to Docs needed? Does Overriding Files and fail2ban section mention one of the deleted Files?

**Review by wernerfred — COMMENTED** [review]
Are Updates to Docs needed? Does Overriding Files and fail2ban section mention one of the deleted Files?

**Comment by casperklein** [comment]
These are only internal changes. From a users perspective, there is no change. The two deleted files are all in `target/`, so only used for image building.

File: Dockerfile [hunk 1]
```diff
@@ -172,9 +172,11 @@ RUN for _ in {1..10}; do su - amavis -c "razor-admin -create" ; sleep 3 ; \
 # ––– Fail2Ban, DKIM & DMARC ––––––––––––––––––––
 # –––––––––––––––––––––––––––––––––––––––––––––––
 
-COPY target/fail2ban/jail.conf /etc/fail2ban/jail.conf
-COPY target/fail2ban/filter.d/postfix-sasl.conf /etc/fail2ban/filter.d/postfix-sasl.conf
-RUN mkdir /var/run/fail2ban
+COPY target/fail2ban/jail.local /etc/fail2ban/jail.local
+RUN ln -s /var/log/mail/mail.log /var/log/mail.log && \
```
Comment (aendeavor) [inline Dockerfile:176]: Just styling: We use

``` DOCKERFILE
RUN \
  do_something_command \
  do_some_more_command
```

i.e. use the `\` first when multi-line `RUN` statements appear:)

**Comment by aendeavor** [comment]
From what I can see: Nice! Just this one style-related change and I'm good :D

File: Dockerfile (hunk 1 above)
Comment (casperklein) [reply Dockerfile:176]: Oh, I must have overlooked that, thx.

Comment (casperklein) [reply Dockerfile:176]: My mistake was, that I looked at this example directly above my change, which seems also not correct then? https://github.com/docker-mailserver/docker-mailserver/blob/34c6e46f68f5de60ac08f815252f4e1632eb78b0/Dockerfile#L167-L169

Comment (casperklein) [reply Dockerfile]: I've fixed both styling issues.

**Review by wernerfred — APPROVED** [review]

**Review by aendeavor — APPROVED** [review]

**Pull request merged**